components/AddMemberModal.tsx

```diff
@@ -1,24 +1,86 @@
+import {
+  AnchorWallet,
+  useAnchorWallet,
+  useConnection,
+} from '@solana/wallet-adapter-react'
 import { FormikErrors, useFormik } from 'formik'
+import { useAppSelector } from '../hooks/useAppSelector'
+import { selectCluster } from '../redux/features/wallet/walletSlice'
+
 import { useRef } from 'react'
+import { FanoutClient } from '@glasseaters/hydra-sdk'
+import { Transaction } from '@solana/web3.js'
+
+type AddMemberModalProps = {
+  wallet: any
+}
 
 interface FormValues {
   pubKey: string
   shares: number
 }
 
-const AddMemberModal = () => {
+const AddMemberModal = ({ wallet }: AddMemberModalProps) => {
   let toggleRef = useRef<HTMLInputElement>(null)
 
   const initialValues = {
     pubKey: '',
-    shares: 0
+    shares: 0,
   }
 
-  const onSubmit = (values: any) => {
+  const { connection } = useConnection()
+  const cluster = useAppSelector(selectCluster)
+  const walletPub: AnchorWallet | any = useAnchorWallet()
+  const onSubmit = async (values: any) => {
     console.log('submitted', values)
     // add the wallet member here
+    if (!wallet) {
+      return
+    }
 
-    toggleRef.current!.checked = false
+    try {
+      const fanoutSdk = new FanoutClient(connection, wallet)
+      // Calculate fanout public key
+      const [fanoutPubkey] = await FanoutClient.fanoutKey('trav')
```

Comment:
You don't need to calculate the fanout public key here, you already have it in the `wallet` object. The reason I did calculate it when creating a wallet is that the user only provides the name of the wallet and I had to derive its public key from that.